tests/test_custom_scalars.py

```diff
@@ -60,7 +67,21 @@ def parse_date_literal(ast):
     return parsed_datetime.date()
 
 
-schema = make_executable_schema(type_defs, [query, datereadonly, dateinput])
+scalar_with_default_parser = ScalarType("ScalarWithDefaultParser")
+
+
+@scalar_with_default_parser.value_parser
+def parse_value_from_default_literal_parser(value):
+    print(value)
+    if not isinstance(value, (str, bool)):
+        raise ValueError()
```

Comment:
I'm currently trying to figure out what the correct method for checking the type of a custom scalar is.
Looks like either you check the type of the value in a value parser or the type of the AST node in a literal parser.
I ended up here while researching that.
Now I wonder why a `ValueError` is raised here. Shouldn't that be a `TypeError` instead or is there something I don't get? Also, what's the proper way for parsing a date out of a string? Checking the AST node type in a literal parser or checking the value type in a value parser?

Reply:
@levrik We are happy to discuss this, but comment to code fragment in merged PR feels quite wrong place to do this. Could you start discussion on [our Spectrum](https://spectrum.chat/ariadne/general/) or new issue? Thanks!